dragonfly/room2d.py

```diff
@@ -1016,6 +1033,62 @@ def intersect_adjacency(room_2ds, tolerance=0.01):
             intersected_rooms.append(rebuilt_room)
         return tuple(intersected_rooms)
 
+    def _honeybee_plenum(self, plenum_ht, tolerance=0.01):
+        """
+        Auto-generate floor and ceiling plenums when converting to a Honeybee Room.
+
+        The boundary condition for all plenum faces is adiabatic except for the
+        ceiling and floor surfaces between the room, and any outdoor walls. Note that the
+        floor and ceiling plenums are modeled individually in this method, so the bottom
+        surface of the floor plenum is not the ceiling of the zone below, and the top
+        surface of the ceiling plenum is not the floor of the zone above.
+
+        Args:
+            hb_room: A honeybee Room representing the dragonfly Room2D.
+
+        Returns:
+            # TODO: Revise this.
+            A honeybee Room mutated to include floor and ceiling plenums.
+        """
+
+        plenums = []
+        plenum_id = self.identifier + '_ceil_plenum'
+        # Create reference 2d geometry for plenums
+        ref_face3d = self.floor_geometry.duplicate()
+        ref_face3d = ref_face3d.move(Vector3D(0, 0, self.floor_to_ceiling_height))
+        # create the honeybee Room
+        ceil_polyface = Polyface3D.from_offset_face(ref_face3d, plenum_ht)
+        plenum_hb_room = Room.from_polyface3d(plenum_id, ceil_polyface)
+
+        # assign BCs and record any Surface conditions to be set on the story level
+        #adjacencies = []
+        for i, bc in enumerate(self._boundary_conditions):
+            if not isinstance(bc, Surface):
+                plenum_hb_room[i + 1].boundary_condition = bc
+            # else:
+            #     adjacencies.append(
+            #         (plenum_hb_room[i + 1], bc.boundary_condition_objects))
```

Comment:
If I am being honest, I was originally thinking that this situation of a Surface boundary condition for a plenum wall would be simpler (as in, we just make it Adiabatic).  This is because most plenums would be adjacent to other plenums and so you would have to figure out what this adjacent plenum's ID is (I guess that's not too difficult if we are using a standard naming convention). I guess if you have it mostly figured out, we can go that route.

Reply:
Maybe to put it another way, I would much rather get this method working correctly for both floor and ceiling plenums than to have it correctly doing this splitting and adjacency setting correctly.  And it just feels like this splitting could be a rabbit hole right now that is going to stop us from reaching this first goal. So I would ask: do you feel that you have this splitting figured out enough to reliably get it working for both floor and ceiling plenums?

Reply:
I agree with you, I think it's better to get the simpler scenario working first and then build on complexity from there. Additionally, if I'm interpreting your comments correctly, and the plenum walls are going to be adiabatic anyway, there isn't a benefit to splitting plenum walls.